### PR TITLE
feat(windows): left-only hotkeys

### LINKED ISSUE #364: [Windows] Distinguish between RAlt and LAlt for shortcuts in Keyman
Distinguish between RAlt and LAlt for shortcuts in Keyman (Windows) and assign shortcuts to SHIFT + LALT sequences, making available SHIFT + RALT key sequences available for keyboard rules.

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #364.

Hotkeys in Windows are currently implemented using RegisterHotkey.
However RegisterHotkey does not distinguish between left and right
modifier keys, which means that they override some keyboards' RAlt
combinations.

This feature reimplements hotkey support in the low level keyboard hook,
and adds a feature flag / debug flag Flag_UseRegisterHotkey to return
to the RegisterHotkey model if that is so needed.

I have made minimal refactoring changes to this code, although there
are certainly functions here which could be improved.

I have flagged a couple of potential future improvements with TODO.

I will not cherry-pick this to 12.0; it's a new feature that can wait for 13.0 release.

**note to reviewer**: Use 'ignore whitespace' to see code changes more clearly.

### CODE REVIEW

File: windows/src/engine/keyman32/selectkeyboard.cpp [hunk 1]
```diff
@@ -179,23 +179,42 @@ void SelectKeyboardHKL(PKEYMAN64THREADDATA _td, DWORD hkl, BOOL foreground) {
     }
 }
 
+void PrepareLanguageSwitchString(UINT langid, HKL hkl, char *str) {
+  wsprintf(str, "%d|%d|%d", GetCurrentThreadId(), langid, hkl);   // I4285
+}
+
+void PrepareLanguageSwitchString(UINT langid, GUID clsid, GUID guidProfile, char *str) {
+  WCHAR clsidstr[40], profilestr[40];
```
Comment (darcywong00) [inline windows/src/engine/keyman32/selectkeyboard.cpp]: Is the 40 character limit tied to anything?

**Comment by darcywong00** [comment]
Is the Keyman Configuration option "Simulate AltGr with Ctrl+Alt" unaffected by this PR?

File: windows/src/engine/keyman/UfrmKeyman7Main.pas [hunk 2]
```diff
@@ -1279,6 +1284,22 @@ procedure TfrmKeyman7Main.WndProc(var Message: TMessage);
...
+  procedure ProcessHotkeyChange(hkl: DWORD; guidProfile: TGUID);
+  var
+    kbd: TLangSwitchKeyboard;
+  begin
+    kbd := FLangSwitchManager.FindKeyboard(hkl, guidProfile);
+    if Assigned(kbd) then
+    begin
+      // Handle toggle hotkey
+      if (kbd = FLangSwitchManager.ActiveKeyboard) and (kmcom.Options['koKeyboardHotkeysAreToggle'].Value) then
+          kbd := FLangSwitchManager.Languages[0].Keyboards[0];
```
Comment (darcywong00) [inline windows/src/engine/keyman/UfrmKeyman7Main.pas]: Ah, this is toggling the Keyman keyboard off into the default keyboard layout.

So if the default keyboard is selected (not `Assigned(kbd)`), will this also toggle into the Keyman keyboard?

File: windows/src/engine/keyman32/selectkeyboard.cpp (hunk 1 above)
Comment (mcdurdin) [reply windows/src/engine/keyman32/selectkeyboard.cpp]: 38 characters in a GUID string returned from StringFromGUID2. https://stackoverflow.com/questions/968175/what-is-the-string-length-of-a-guid (brace format)

**Comment by mcdurdin** [comment]
> Is the Keyman Configuration option "Simulate AltGr with Ctrl+Alt" unaffected by this PR?

Yes, that is a separate code path. There may be conflicts with hotkeys and these keys but given this option is needed rarely, I think users can disable conflicting hotkeys in this instance.

File: windows/src/engine/keyman/UfrmKeyman7Main.pas (hunk 2 above)
Comment (mcdurdin) [reply windows/src/engine/keyman/UfrmKeyman7Main.pas]: `FindKeyboard` returns `nil` (i.e. `not Assigned`) if no keyboard can be found that matches the hotkey. Even if the default keyboard is currently selected, `FindKeyboard` will still return the keyboard associated with hotkey, so the default keyboard won't be selected

**Pull request merged**